sdk/typespec/src/error/mod.rs

```diff
@@ -169,6 +169,14 @@ impl Error {
         }
     }
 
+    #[cfg(feature = "http")]
+    pub fn http_status(&self) -> Option<StatusCode> {
```

Comment:
I think for now this is fine, but for posterity I question whether this is too specific at least for HTTP status code. I could imagine returning more since we're already declaring it - e.g., a tuple of the status code and status message - but have no basis to suggest any changes now. We'll watch feedback.

Reply:
The main reason I added this was to be able to `match` against an HTTP error here:

https://github.com/Azure/azure-sdk-for-rust/blob/1f2cef7c74cc0c8fb423986544cfad472cd517da/sdk/cosmos/azure_data_cosmos/examples/cosmos/read.rs#L33-L44

Being able to create concise match arms against HTTP status codes is pretty useful as a consumer of our APIs. I'm not 100% happy with this yet, but it's a start